File: Samples/Draw and print PDF/PrintPdf/C#/MainForm.cs
using System;
using System.Windows.Forms;
using BitMiracle.Docotic.Pdf;

namespace BitMiracle.Docotic.Samples.PrintPdf
{
    public partial class MainForm : Form
    {
        private const int FitPageIndex = 0;

        public MainForm()
        {
            InitializeComponent();
        }

        private PrintSize getPrintSize()
        {
            return (printSize.SelectedIndex == FitPageIndex) ? PrintSize.FitPage : PrintSize.ActualSize;
        }

        private void printButton_Click(object sender, EventArgs e)
        {
            processExistingPdfDocument(PdfPrintHelper.ShowPrintDialog);
        }

        private void previewButton_Click(object sender, EventArgs e)
        {
            processExistingPdfDocument(PdfPrintHelper.ShowPrintPreview);
        }

        private void processExistingPdfDocument(Func<PdfDocument, PrintSize, DialogResult> action)
        {
            using (OpenFileDialog dlg = new OpenFileDialog())
            {
                dlg.Filter = "PDF files (*.pdf)|*.pdf";

                if (dlg.ShowDialog() == DialogResult.OK)
                {
                    // NOTE: 
                    // When used in trial mode, the library imposes some restrictions.
                    // Please visit http://bitmiracle.com/pdf-library/trial-restrictions.aspx
                    // for more information.

                    using (PdfDocument pdf = new PdfDocument(dlg.FileName))
                        action(pdf, getPrintSize());
                }
            }
        }
    }
}

File: Samples/Draw and print PDF/PrintPdf/VB.NET/MainForm.vb
Imports System
Imports System.Windows.Forms
Imports BitMiracle.Docotic.Pdf

Namespace BitMiracle.Docotic.Samples.PrintPdf
    Public Partial Class MainForm
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub printButton_Click(ByVal sender As Object, ByVal e As EventArgs)
            processExistingPdfDocument(New Action(Of PdfDocument)(AddressOf ShowPrintDialog))
        End Sub

        Private Sub previewButton_Click(ByVal sender As Object, ByVal e As EventArgs)
            processExistingPdfDocument(New Action(Of PdfDocument)(AddressOf ShowPrintPreview))
        End Sub

        Private Sub processExistingPdfDocument(ByVal action As Action(Of PdfDocument))
            Using dlg As OpenFileDialog = New OpenFileDialog()
                dlg.Filter = "PDF files (*.pdf)|*.pdf"

                If dlg.ShowDialog() = DialogResult.OK Then

                    Using pdf As PdfDocument = New PdfDocument(dlg.FileName)
                        action(pdf)
                    End Using
                End If
            End Using
        End Sub
    End Class
End Namespace
